lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,6 +155,26 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        include ::TrilogyAdapter::BackwardsCompatibility
+      else # ActiveRecord >= 7.1 support
+        alias_method :with_trilogy_connection, :with_raw_connection
+      end
+
+      alias_method :reset!, :reconnect!
+
+      def raw_execute(sql, name, async: false, allow_retry: false, uses_transaction: true)
```

Comment:
We're relying on the upstream `raw_execute` since https://github.com/github/activerecord-trilogy-adapter/pull/29. I think we should keep doing that so we don't fall out of sync (e.g. the version here is missing the new warnings feature).